rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -98,21 +110,164 @@ def _decode_lido_staking_in_steth(self, context: DecoderContext, sender: Checksu
 
         return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO)
 
+    def _decode_deposit_event(self, context: DecoderContext, depositor: ChecksumEvmAddress) -> DecodingOutput:
+        minted_wsteth_amount_raw = int.from_bytes(context.tx_log.data[:32])
+        minted_wsteth_amount = asset_normalized_value(
+            amount=minted_wsteth_amount_raw,
+            asset=A_WSTETH.resolve_to_crypto_asset(),
+        )
+
+        # A steth -> wsteth wrapping event has 4 logs associated with it.
+        #
+        # 1. A transfer of wsteth to the user
+        # 2. An approval of steth to reduce the wsteth contract's allowance
+        # 3. A transfer of steth tokens from the user to the wsteth contract
+        # 4. A transfer of steth shares from the user to the wsteth contract
+        #
+        # We're currently decoding the ERC20 transfer associated with the first log
+        # so we can find the other 3 logs and pull values from them accordingly.
+
+        tx_start_log_index = context.all_logs[0].log_index
+        steth_transfer_log_index = (context.tx_log.log_index - tx_start_log_index) + 2
+        steth_transfer_log = context.all_logs[steth_transfer_log_index]
```

Comment:
Here I want to index `all_logs` related to the log I'm currently decoding. I imagine this is a common requirement so is there an existing helper method to do this?